### PR TITLE
Feature/permission manager bulk

### PR DESCRIPTION
**Pull request by novaknole**
Adds a new feature for bulk in order to set permissions on multiple targets..

### CODE REVIEW

**Review by Rekard0 — DISMISSED** [review]
LGTM!

**Review by Michael-A-Heuer — APPROVED** [review]

File: packages/contracts/contracts/core/permission/BulkPermissionsLib.sol [hunk 1]
```diff
@@ -9,12 +9,22 @@ library BulkPermissionsLib {
     enum Operation {
         Grant,
         Revoke,
-        Freeze
+        Freeze,
+        GrantWithOracle
     }
 
-    struct Item {
+    struct ItemSingleTarget {
```
Comment (brickpop) [inline packages/contracts/contracts/core/permission/BulkPermissionsLib.sol:16]: BulkPermissionSingle

File: packages/contracts/contracts/core/permission/BulkPermissionsLib.sol [hunk 2]
```diff
@@ -9,12 +9,22 @@ library BulkPermissionsLib {
...
         Operation operation;
+        address who;
         bytes32 permissionId;
+    }
+
+    struct ItemMultiTarget {
```
Comment (brickpop) [inline packages/contracts/contracts/core/permission/BulkPermissionsLib.sol:22]: BulkPermissionMulti

File: packages/contracts/contracts/core/permission/PermissionManager.sol [hunk 3]
```diff
@@ -166,12 +157,12 @@ contract PermissionManager is Initializable {
     /// @dev Requires the `ROOT_PERMISSION_ID` permission.
     /// @param _where The address of the contract.
     /// @param items The array of bulk items to process.
-    function bulk(address _where, BulkPermissionsLib.Item[] calldata items)
+    function bulkOnSingleTarget(address _where, BulkPermissionsLib.ItemSingleTarget[] calldata items)
```
Comment (brickpop) [inline packages/contracts/contracts/core/permission/PermissionManager.sol:160]: `applySingleBulk`

File: packages/contracts/contracts/core/permission/PermissionManager.sol [hunk 4]
```diff
@@ -182,7 +173,34 @@ contract PermissionManager is Initializable {
         }
     }
 
-    /// @notice Checks if a permission is granted to an address on a contract via a permission identifier and considers if `ANY_ADDRESS` was used in the granting process.
+    /// @notice Processes bulk items on the permission manager.
+    /// @dev Requires that msg.sender has each permissionId on the where.
+    /// @param items The array of bulk items to process.
+    function bulkOnMultiTarget(BulkPermissionsLib.ItemMultiTarget[] calldata items) external {
```
Comment (brickpop) [inline packages/contracts/contracts/core/permission/PermissionManager.sol:179]: `applyMultiBulk`

File: packages/contracts/test/core/permission/permission-manager.ts [hunk 5]
```diff
@@ -14,16 +14,27 @@ const ALLOW_FLAG = ethers.utils.getAddress(
...
+
 enum Operation {
   Grant,
   Revoke,
   Freeze,
+  GrantWithOracle
 }
 
-interface Item {
+interface ItemSingleTarget {
```
Comment (brickpop) [inline packages/contracts/test/core/permission/permission-manager.ts:26]: Rename accordingly

File: packages/contracts/test/core/permission/permission-manager.ts [hunk 6]
```diff
@@ -14,16 +14,27 @@ const ALLOW_FLAG = ethers.utils.getAddress(
...
   operation: Operation;
+  who: string;
   permissionId: string;
+}
+
+interface ItemMultiTarget {
```
Comment (brickpop) [inline packages/contracts/test/core/permission/permission-manager.ts:32]: Rename accordingly

**Review by brickpop — COMMENTED** [review]
Code looks good, but I suggest a follow-up PR to use more meaningful names

**Pull request merged**